dGame/dComponents/PetComponent.cpp

```diff
@@ -162,86 +162,40 @@ void PetComponent::OnUse(Entity* originator) {
 		return;
 	}
 
-	auto* movementAIComponent = m_Parent->GetComponent<MovementAIComponent>();
-
+	auto* const movementAIComponent = m_Parent->GetComponent<MovementAIComponent>();
 	if (movementAIComponent != nullptr) {
 		movementAIComponent->Stop();
 	}
 
 	inventoryComponent->DespawnPet();
 
-	const auto& cached = buildCache.find(m_Parent->GetLOT());
-	int32_t imaginationCost = 0;
-
-	std::string buildFile;
-
-	if (cached == buildCache.end()) {
-		auto query = CDClientDatabase::CreatePreppedStmt(
-			"SELECT ValidPiecesLXF, PuzzleModelLot, Timelimit, NumValidPieces, imagCostPerBuild FROM TamingBuildPuzzles WHERE NPCLot = ?;");
-		query.bind(1, static_cast<int>(m_Parent->GetLOT()));
-
-		auto result = query.execQuery();
-
-		if (result.eof()) {
-			ChatPackets::SendSystemMessage(originator->GetSystemAddress(), u"Failed to find the puzzle minigame for this pet.");
-
-			return;
-		}
-
-		if (result.fieldIsNull("ValidPiecesLXF")) {
-			result.finalize();
-
-			return;
-		}
-
-		buildFile = std::string(result.getStringField("ValidPiecesLXF"));
-
-		PetPuzzleData data;
-		data.buildFile = buildFile;
-		data.puzzleModelLot = result.getIntField("PuzzleModelLot");
-		data.timeLimit = result.getFloatField("Timelimit");
-		data.numValidPieces = result.getIntField("NumValidPieces");
-		data.imaginationCost = result.getIntField("imagCostPerBuild");
-		if (data.timeLimit <= 0) data.timeLimit = 60;
-		imaginationCost = data.imaginationCost;
-
-		buildCache[m_Parent->GetLOT()] = data;
-
-		result.finalize();
-	} else {
-		buildFile = cached->second.buildFile;
-		imaginationCost = cached->second.imaginationCost;
-	}
-
-	auto* destroyableComponent = originator->GetComponent<DestroyableComponent>();
+	const auto& entry =  CDClientManager::GetTable<CDTamingBuildPuzzleTable>()->GetByLOT(m_Parent->GetLOT());
 
+	const auto* const destroyableComponent = originator->GetComponent<DestroyableComponent>();
 	if (destroyableComponent == nullptr) {
 		return;
 	}
 
-	auto imagination = destroyableComponent->GetImagination();
-
-	if (imagination < imaginationCost) {
+	const auto imagination = destroyableComponent->GetImagination();
+	if (imagination < entry.imaginationCost) {
 		return;
 	}
 
-	const auto& bricks = BrickDatabase::GetBricks(buildFile);
-
+	const auto& bricks = BrickDatabase::GetBricks(entry.validPieces);
 	if (bricks.empty()) {
 		ChatPackets::SendSystemMessage(originator->GetSystemAddress(), u"Failed to load the puzzle minigame for this pet.");
-		LOG("Couldn't find %s for minigame!", buildFile.c_str());
+		LOG("Couldn't find %s for minigame!", entry.validPieces.c_str());
```

Comment:
```suggestion
		LOG("Couldn't find %s for minigame!", entry.validPieces.string().c_str());
```

Reply:
Not needed as far as I can tell. Path has its own c_string() method I think.

Reply:
using paths is dumb here and I went back to strings